### PR TITLE
Make stats command delete if used in server

### PR DESCRIPTION
**Pull request by MoonlightCapital**
# Please check the following boxes

- [x] I've read and agree to the [Contribution Guidelines](https://github.com/Dougley/MBv2/blob/master/.github/CONTRIBUTING.md) and to the [Code of Conduct](https://github.com/Dougley/MBv2/blob/master/.github/CODE_OF_CONDUCT.md)
- [x] I've checked that I didn't commit sensitive information like tokens or other login information
- [x] I haven't hardcoded any IDs or anything else that shouldn't be (with the exception being permissions)
- [x] I tested my code and I verified it's working to the best of my ability
- [x] I checked if my code doesn't violate the styleguide with `npm test`

# Describe your pull request

Reduces spam in server channels

### CODE REVIEW

File: src/commands/stats.js [hunk 1]
```diff
@@ -7,9 +7,11 @@ module.exports = {
   },
   fn: async (msg) => {
     const data = db.getUser(msg.author.id)
-    msg.author.getDMChannel().then(c => {
+    msg.author.getDMChannel().then(async c => {
       if (data.entitlements.includes('fake-stats')) msg.channel.createMessage(`<@${msg.author.id}>, an unexpected error occurred while getting your stats, try again later.`)
-      return c.createMessage(generateEmbed(msg.author, data)).then(() => { if (!data.entitlements.includes('fake-stats')) msg.addReaction(`${ids.emojis.confirm.name}:${ids.emojis.confirm.id}`) })
+      await c.createMessage(generateEmbed(msg.author, data))
+      if (msg.channel.guild) return msg.delete()
+      if (!data.entitlements.includes('fake-stats')) await msg.addReaction(`${ids.emojis.confirm.name}:${ids.emojis.confirm.id}`)
```
Comment (Dougley) [inline src/commands/stats.js]: You're removing the message, why react to it?
```suggestion
```

File: src/commands/stats.js [hunk 2]
```diff
@@ -7,9 +7,11 @@ module.exports = {
   },
   fn: async (msg) => {
```
Comment (Dougley) [inline src/commands/stats.js]: Top level async is redundant at this point
```suggestion
   fn: (msg) => {
```

File: src/commands/stats.js (hunk 1 above)
Comment (MoonlightCapital) [reply src/commands/stats.js]: That happens after the return if the message is sent in a server.

Comment (Dougley) [reply src/commands/stats.js]: 👍

**Comment by Dougley** [comment]
Resolve your style issues and I'll merge this 👌

**Pull request merged**